File: 2609-find-the-longest-balanced-substring-of-a-binary-string/2609-find-the-longest-balanced-substring-of-a-binary-string.js
/**
 * @param {string} s
 * @return {number}
 */
var findTheLongestBalancedSubstring = function(s) {
    let ones=0
    let zeros=0
    let max=0
    for(let i=0;i<s.length;i++){
        if(s[i]==='0'){
            if(ones!==0){
                ones=0
                zeros=0
            }
            
            zeros+=1
        }else{
            ones+=1
            if(zeros!==0){
                max=Math.max(max,2*Math.min(ones,zeros))
            }
        }
    }
    return max
};

File: 2609-find-the-longest-balanced-substring-of-a-binary-string/2609-find-the-longest-balanced-substring-of-a-binary-string.py
class Solution:
    def findTheLongestBalancedSubstring(self, s: str) -> int:
        zeros=0
        ones=0
        maxx=0
        for i in range(len(s)):
            if s[i]=='0':
                if ones!=0:
                    zeros=0
                    ones=0
                zeros+=1
            else:
                ones+=1
                if zeros!=0:
                    maxx=max(maxx,2*min(ones,zeros))
        return maxx
        